### ISSUE TITLE
Samsung Contacts provide phonenumbers with space, cant sanitize number for query search

### ISSUE THREAD

**Issue by wo-ist-henry** [vestrel00/contacts-android#196]
Hi everyone,

when comparing phonenumbers, the given `normalizedNumber` is in my cases null and I'm not able to sanitize the given numbers in the where clause.

Can sanitizing on my own be achieved or is there a fix in the library needed?

Besides that your library works flawless!

```
 suspend fun queryLocalSingleContact(searchText: String): Contact? =
        Contacts(applicationContext)
            .query()
            .where {
                ( Phone.normalizedNumber equalTo searchText)
            }
            .include(contactFieldsToInclude())
            .orderBy(ContactsFields.LastUpdatedTimestamp.desc())
            .limit(1)
            .findWithContext()
            .firstOrNull()
```

**Comment by wo-ist-henry** [comment]
Found my mistake.

You have to use the `broadQuery()` with ` .whereAnyContactDataPartiallyMatches(searchText)`

**Comment by vestrel00** [comment]
@wo-ist-henry thanks for creating this issue! You can certainly use `broadQuery()` with `.whereAnyContactDataPartiallyMatches(searchText)`. However, if you are looking for contacts that match a specific phone number, then the current `BroadQuery` API may not return what you need.

For example, the query `.whereAnyContactDataPartiallyMatches("718")` **could** match contacts that don't have matching phone numbers. It will match contacts with an address, email, organization, note, etc that contains "718" even if the contact does not have a phone number.

Therefore, I create issue #197 to add support in `BroadQuery` to match only phone or email only. It will be included in the next release [(0.2.0)](https://github.com/vestrel00/contacts-android/discussions/146).

Once I make the code changes, your code should look like,

```kotlin
suspend fun queryLocalSingleContact(searchText: String): Contact? =
        Contacts(applicationContext)
            .broadQuery()
            .match(BroadQuery.Match.PHONE)
            .wherePartiallyMatches(searchText)
            .include(contactFieldsToInclude())
            .orderBy(ContactsFields.LastUpdatedTimestamp.desc())
            .limit(1)
            .findWithContext()
            .firstOrNull()
```

If you are looking to match any contact data with your `searchText`, then the current API will give you the same exact search results as the AOSP Contacts app and Google Contacts app 😁 

> when comparing phonenumbers, the given normalizedNumber is in my cases null and I'm not able to sanitize the given numbers in the where clause.
> 
> Can sanitizing on my own be achieved or is there a fix in the library needed?

As you already know, database values for `Phone.normalizedNumber` can be null especially if the `Phone.number` is invalid or not recognized to be a proper format by the system and locale. This is the same behavior as with the AOSP Contacts app and Google Contacts app and is not exclusive to this library. It is not a bug with the library. 

You can certainly use `android.telephony.PhoneNumberUtils.formatNumberToE164` to sanitize/convert the `Phone.number` to `Phone.normalizedNumber`. However, that function can return null if the given phone number is not valid. The Contacts Provider most likely uses the same functions automatically whenever a phone is inserted or update into the DB.

As for using the `Query` API, it will not work if you want to match contacts with phone numbers using a free-form search string because `normalizedNumber` can be null.
